### PR TITLE
[WIP] ci: debug Python tests

### PR DESCRIPTION
**Pull request by blueyed**
The aim of this PR is to get the python oldtests run on CI - `Test_vim_function` is expected to fail there.

### CODE REVIEW

**Comment by justinmk** [comment]
What problem is being debugged here? Python tests are not a high priority right now.

File: ci/build.ps1 [hunk 1]
```diff
@@ -111,8 +111,8 @@ cmake --build . --config $cmakeBuildType -- $cmakeGeneratorArgs ; exitIfFailed
 .\bin\nvim -u NONE --headless -c 'exe !has(\"win32\").\"cq\"' ; exitIfFailed
 
 # Ensure Python is available.
-.\bin\nvim -u NONE -c 'exe !has("python")."cq"' ; exitIfFailed
-.\bin\nvim -u NONE -c 'exe !has("python3")."cq"' ; exitIfFailed
+.\bin\nvim -u NONE -c 'exe !has(\"python\").\"cq\"' ; exitIfFailed
+.\bin\nvim -u NONE -c 'exe !has(\"python3\").\"cq\"' ; exitIfFailed
```
Comment (justinmk) [inline ci/build.ps1]: AFAIR the powershell escape char is backtick.

File: ci/run_tests.sh [hunk 2]
```diff
@@ -8,14 +8,20 @@ source "${CI_DIR}/common/build.sh"
 source "${CI_DIR}/common/test.sh"
 source "${CI_DIR}/common/suite.sh"
 
+# Build.
 enter_suite build
```
Comment (justinmk) [inline ci/run_tests.sh]: isn't "enter suite ..." already descriptive enough?

File: src/nvim/testdir/test_python3.vim [hunk 3]
```diff
@@ -2,7 +2,7 @@
 " TODO: move tests from test88.in here.
 
 if !has('python3')
-  finish
+  throw 'Skipped: does not have +python3'
```
Comment (justinmk) [inline src/nvim/testdir/test_python3.vim]: `+python{2,3}` never exists. The feature is called `python{2,3}` (without `+`). Full list of features is at `:h feature-list` (well, it's missing "ruby"...).

**Comment by justinmk** [comment]
see https://github.com/neovim/neovim/issues/10270#issuecomment-503682823

File: ci/build.ps1 (hunk 1 above)
Comment (blueyed) [reply ci/build.ps1]: This should be OK - IIRC I've tried the backtick also back with the has-win32 check/PR (recently merged).

File: ci/run_tests.sh (hunk 2 above)
Comment (blueyed) [reply ci/run_tests.sh]: I found it hard to visualize the different sections therein - so it is more meant as a section headline.

File: src/nvim/testdir/test_python3.vim (hunk 3 above)
Comment (blueyed) [reply src/nvim/testdir/test_python3.vim]: Yes, that's only for naming, but I agree that is confusing - only temporary though.

I do not get why it is not available during tests, but before - i.e. the tests gets skipped, but I'd expect them to run (and fail).

**Comment by blueyed** [comment]
`:checkhealth` reports:
```
## Python 2 provider (optional)
  - INFO: pyenv: Path: /opt/pyenv/libexec/pyenv
  - INFO: pyenv: Root: /opt/pyenv
  - INFO: `g:python_host_prog` is not set.  Searching for python2 in the environment.
  - WARNING: pyenv is not set up optimally.
    - ADVICE:
      - Create a virtualenv specifically for Neovim using pyenv, and set `g:python_host_prog`.  This will avoid the need to install the pynvim module in each version/virtualenv.
  - INFO: Executable: /opt/pyenv/versions/2.7.15/bin/python2
  - INFO: Python version: 2.7.15
  - INFO: pynvim version: 0.3.2
  - OK: Latest pynvim is installed.
## Python 3 provider (optional)
  - INFO: pyenv: Path: /opt/pyenv/libexec/pyenv
  - INFO: pyenv: Root: /opt/pyenv
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - WARNING: pyenv is not set up optimally.
    - ADVICE:
      - Create a virtualenv specifically for Neovim using pyenv, and set `g:python3_host_prog`.  This will avoid the need to install the pynvim module in each version/virtualenv.
  - INFO: Executable: /opt/pyenv/versions/3.7/bin/python3
  - INFO: Python version: 3.7.1
  - INFO: pynvim version: 0.3.2
  - OK: Latest pynvim is installed.
```

But `has('python{,3}')` is still false with oldtests?!

**Comment by justinmk** [comment]
pyenv is messing it up. Why are we using pyenv?

**Comment by blueyed** [comment]
> pyenv is messing it up.

How?
Note also that `build/bin/nvim -u NONE -c 'exe !has("python")."cq"' || { echo "Python 2 is not available"; exit 1;}` works (on Travis).

> Why are we using pyenv?

Because Travis uses it mostly.  Added the osx job, which does not use it.

**Comment by blueyed** [comment]
Mystery solved: https://github.com/neovim/neovim/issues/10270#issuecomment-531516948

**Comment by blueyed** [comment]
JFI:

> pyenv is messing it up. Why are we using pyenv?

I think using pyenv (manually) actually allows us to install it for the used Python, which would not be the case if using the system Python (without "sudo" as least).

**Pull request closed without merging**